lib/marked.js

```diff
@@ -1094,7 +1094,8 @@ function escape(html, encode) {
 }
 
 function unescape(html) {
-  return html.replace(/&([#\w]+);/g, function(_, n) {
+	// explicitly match decimal, hex, and named HTML entities 
+  return html.replace(/&(#(?:\d+)|(?:#x[0-9A-Fa-f]+)|(\w+))/g, function(_, n) {
```

Comment:
@matt- Shouldn't there be an optional semicolon in the regex so that unescape wouldn't leave the semicolons in the string if they are there?
I mean something like this:
`/&(#(?:\d+)|(?:#x[0-9A-Fa-f]+)|(\w+));?/g`
instead of:
`/&(#(?:\d+)|(?:#x[0-9A-Fa-f]+)|(\w+))/g`
or possibly with the third non-capturing group to be consistent with the first two:
`/&(#(?:\d+)|(?:#x[0-9A-Fa-f]+)|(?:\w+));?/g`